include/swift/Basic/Casting.h

```diff
@@ -0,0 +1,30 @@
+//===--- Casting.h - Helpers for casting ------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_CASTING_H
+#define SWIFT_BASIC_CASTING_H
+
+#include "STLCompatibility.h"
```

Comment:
Can we inline the `std::bit_cast` implementation for compatibility with older C++ standards?